apps/rush-lib/src/logic/policy/LockfilePolicy.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as colors from 'colors';
+import * as os from 'os';
+import { InternalError } from '@microsoft/node-core-library';
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { AlreadyReportedError } from '../../utilities/AlreadyReportedError';
+import { IPolicyValidatorOptions } from './PolicyValidator';
+import { BaseShrinkwrapFile } from '../base/BaseShrinkwrapFile';
+import { ShrinkwrapFileFactory } from '../ShrinkwrapFileFactory';
+
+export class LockfilePolicy {
+  public static validate(rushConfiguration: RushConfiguration, options: IPolicyValidatorOptions): void {
+    // If allowShrinkwrapUpdates is true, then we're updating the shrinkwrap
+    // and cannot validate the hash
+    if (options.allowShrinkwrapUpdates) {
+      return;
+    }
+
+    if (!LockfilePolicy.isEnabled(rushConfiguration)) {
+      // If the option wasn't provided
+      console.log(
+        colors.cyan(
+          'Ignoring lockfile validation because it was not enabled or is not supported ' +
+          'with the specified package manager.'
+        ) + os.EOL
+      );
+      return;
+    }
+
+    console.log('Checking shrinkwrap hash to validate no manual changes were made.' + os.EOL);
```

Comment:
We really should implement a "verbose logging" feature for Rush.  Otherwise, this also feels like noise.

Reply:
Oh yeah I absolutely want to do that. I started switching Rush over to use the `Terminal` API a while ago but never finished it.